bot/exts/moderation/infraction/infractions.py

```diff
@@ -664,6 +645,26 @@ async def cog_command_error(self, ctx: Context, error: Exception) -> None:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """
+
+        Apply active timeout infractions for returning members.
+
+        This is needed for users who might have had their infraction edited in our database but not in Discord itself.
+        """
+        active_timeouts = await self.bot.api_client.get(
+            endpoint="bot/infractions",
+            params={"active": "true", "type": "timeout", "user__id": member.id}
+        )
+
+        if active_timeouts and not member.is_timed_out():
+            reason = f"Applying active timeout for returning member: {active_timeouts[0]['id']}"
+
+            async def action() -> None:
+                await member.edit(timed_out_until=arrow.get(active_timeouts[0]["expires_at"]).datetime, reason=reason)
+            await self.reapply_infraction(active_timeouts[0], action)
```

Comment:
Doesn't this overlook users whose timeout duration was edited while they were not present on the server but are still under a timeout when they rejoin the server? Once they rejoin the server, they will still be under the old timeout duration and not the new edited duration.

Reply:
How does [this](https://github.com/python-discord/bot/compare/36ee066ff883d56ef31d129a45d37e35a18849ab..8e1ff85c94afd86e6aee42de2ea7c3fdc995938e) look